lib/smart_proxy_remote_execution_ssh/api.rb

```diff
@@ -4,6 +4,98 @@ class Api < ::Sinatra::Base
       get "/pubkey" do
         File.read(Ssh.public_key_file)
       end
+      post "/session" do
+        if env["HTTP_CONNECTION"] != "upgrade" or env["HTTP_UPGRADE"] != "raw" then
+          return [ 401, "Invalid request: /ssh/session requires connection upgrade to 'raw'" ]
+        end
+        socket = nil
+        if env['rack.hijack?'] then
+          env['rack.hijack'].call
+          socket = env['rack.hijack_io']
+        end
+        if !socket then
+          return [ 501, "Internal error: request hijacking not available" ]
+        end
+
+        params = MultiJson.load(env["rack.input"].read)
+        key_file = ForemanRemoteExecutionCore.settings.fetch(:ssh_identity_key_file)
+
+        methods = %w(publickey)
+        methods.unshift('password') if params["ssh_password"]
+
+        ssh_options = { }
+        ssh_options[:port] = params["ssh_port"] if params["ssh_port"]
+        ssh_options[:keys] = [ key_file ] if key_file
+        ssh_options[:password] = params["ssh_password"] if params["ssh_password"]
+        ssh_options[:passphrase] = params[:ssh_key_passphrase] if params[:ssh_key_passphrase]
+        ssh_options[:keys_only] = true
+        ssh_options[:auth_methods] = methods
+        ssh_options[:verify_host_key] = :accept_new_or_local_tunnel
```

Comment:
We should also accept a host key as a parameter and setup a suitable known_hosts file with it.